pkg/kubelet/kubelet_pods.go

```diff
@@ -425,7 +425,15 @@ func truncatePodHostnameIfNeeded(podName, hostname string) (string, error) {
 }
 
 // GetOrCreateUserNamespaceMappings returns the configuration for the sandbox user namespace
-func (kl *Kubelet) GetOrCreateUserNamespaceMappings(pod *v1.Pod) (*runtimeapi.UserNamespace, error) {
+func (kl *Kubelet) GetOrCreateUserNamespaceMappings(pod *v1.Pod, runtimeHandler string) (*runtimeapi.UserNamespace, error) {
+	if kl.runtimeHandlers == nil {
+		return nil, fmt.Errorf("runtime handlers are not set")
+	}
+	if h, found := kl.runtimeHandlers[runtimeHandler]; !found {
+		return nil, fmt.Errorf("the handler %q is not known", runtimeHandler)
+	} else if !h.SupportsUserNamespaces {
+		return nil, fmt.Errorf("the handler %q does not support user namespaces", runtimeHandler)
+	}
```

Comment:
Don't we want only to check it if the pod is using userns? Otherwise, wouldn't this fail any pod (even thoes that don't use userns) if the runtime doesn't support userns?

Reply:
This is fixed now